Src/coffeescript/cql-execution/src/elm/aggregate.coffee

```diff
@@ -1,24 +1,32 @@
 { Expression } = require './expression'
 { typeIsArray , allTrue, anyTrue, compact, numerical_sort} = require '../util/util'
 { build } = require './builder'
+{ Exception } = require '../datatypes/exception'
 Quantity = require './quantity'
 
 quantitiesOrArg = (arr) ->
-  if arr[0]?.constructor.name == "Quantity"
+  arr = compact(arr)
+  # short curcuit empty arrays and return
+  if arr.length == 0
+    return arr
+
+  allQs = arr.every (x) -> x.constructor.name == "Quantity"
+  someQs = arr.some (x) -> x.constructor.name == "Quantity"
+  if allQs
     unit = arr[0].unit
     values = []
     for i in arr
-      if i.constructor.name == "Quantity" && i.unit == unit
-        values.push i.value
-      else
-        return []
+      values.push i.convertUnits(unit)
     return compact(values) # need to make sure that there are not any null values from the quntities
```

Comment:
That is what I was assuming, just because the initial array has been compacted I wasn't sure that there would be values wouldn't contain a null.  Though, this is also from 2 years ago before I started throwing exceptions, so I'm not sure the assumption is or isnt valid.  I can take it out if needed.